### PR TITLE
Enhance query tool clarity and prevent unnecessary unsaved changes prompt. #8010 #8127

### PR DESCRIPTION
**Pull request by RohitBhati8269**
1. Improve query tool data output and messages for clarity. [Detail info](https://github.com/pgadmin-org/pgadmin4/issues/8010)
2. Prevent unsaved changes prompt when no changes are made in the query tool. [Detail info](https://github.com/pgadmin-org/pgadmin4/issues/8127)

### CODE REVIEW

File: web/pgadmin/tools/sqleditor/static/js/components/sections/ResultSet.jsx [hunk 1]
```diff
@@ -753,6 +753,7 @@ export class ResultSetUtils {
       }
       this.eventBus.fireEvent(QUERY_TOOL_EVENTS.SET_MESSAGE, tabMsg, true);
       this.eventBus.fireEvent(QUERY_TOOL_EVENTS.FOCUS_PANEL, PANELS.MESSAGES);
+      this.eventBus.fireEvent(QUERY_TOOL_EVENTS.SET_QUERY_DATA, null);
```
Comment (adityatoshniwal) [inline web/pgadmin/tools/sqleditor/static/js/components/sections/ResultSet.jsx:756]: What is this for, no comments added? And why do we need an event in the same component, which is not used in any other component?

File: web/pgadmin/tools/sqleditor/static/js/components/sections/ResultSet.jsx [hunk 2]
```diff
@@ -1046,6 +1047,10 @@ export function ResultSet() {
       setSelectedRows(new Set());
       setAllRowsSelect('NONE');
     });
+
+    eventBus.registerListener(QUERY_TOOL_EVENTS.SET_QUERY_DATA, (data) => {
+      setQueryData(data);
```
Comment (adityatoshniwal) [inline web/pgadmin/tools/sqleditor/static/js/components/sections/ResultSet.jsx:1052]: You can directly call this.

**Review by adityatoshniwal — CHANGES_REQUESTED** [review]

File: web/pgadmin/tools/sqleditor/static/js/components/sections/ResultSet.jsx (hunk 1 above)
Comment (RohitBhati8269) [reply web/pgadmin/tools/sqleditor/static/js/components/sections/ResultSet.jsx:756]: In today's scrum call I discussed with the team about this issue Maybe you left the call.
It is a bug like:-
when we execute any `select` SQL it fetches the data and shows it in the data output tab
And then if we run `commit or alter`  SQL, it shows us a message, but still the data-output tab has the old `select` query value.
That's why I have made these changes. Currently, we are not clearing the data-output tab if a query does not have any result to display.

File: web/pgadmin/tools/sqleditor/static/js/components/sections/ResultSet.jsx (hunk 2 above)
Comment (RohitBhati8269) [reply web/pgadmin/tools/sqleditor/static/js/components/sections/ResultSet.jsx:1052]: Yes, I can call it directly as a prop.
But I thought if in future we can use that event.
Let me know if we don't want this event and I will call it directly.

File: web/pgadmin/tools/sqleditor/static/js/components/sections/ResultSet.jsx (hunk 1 above)
Comment (adityatoshniwal) [reply web/pgadmin/tools/sqleditor/static/js/components/sections/ResultSet.jsx:756]: It was discussed in a call, but how will someone know when reading the code? Add some comments.

File: web/pgadmin/tools/sqleditor/static/js/components/sections/ResultSet.jsx [hunk 3]
```diff
@@ -713,7 +713,7 @@ export class ResultSetUtils {
     return null;
   }
 
-  queryFinished(httpMessage, onResultsAvailable, onExplain) {
+  queryFinished(httpMessage, onResultsAvailable, onExplain, setQueryData) {
```
Comment (adityatoshniwal) [inline web/pgadmin/tools/sqleditor/static/js/components/sections/ResultSet.jsx:716]: why not use `onResultsAvailable` to setQueryData?

**Review by adityatoshniwal — CHANGES_REQUESTED** [review]

**Pull request merged**